Using this PR:

```
$> spack install py-ipykernel
...
==> Installing py-ipykernel-5.3.4-wyn6znqdqfrlv6wklwntnensz4ycpoy3
==> Using cached archive: /opt/spack/var/spack/cache/_source-cache/archive/9b/9b2652af1607986a1b231c62302d070bc0534f564c393a5d9d130db9abbbe89d.tar.gz
==> py-ipykernel: Executing phase: 'build'
==> py-ipykernel: Executing phase: 'install'
==> py-ipykernel: Executing phase: 'install_data'
==> Error: TypeError: expected string or bytes-like object

/opt/spack/var/spack/repos/builtin/packages/py-ipykernel/package.py:48, in install_data:
         46    def install_data(self, spec, prefix):
         47        """ install the Jupyter kernel spec """
  >>     48        self.spec['python'].command('-m ipykernel', ['install'])
...
```

Here's the trace:
```
Traceback (most recent call last):
  File "/opt/spack/lib/spack/spack/build_environment.py", line 847, in _setup_pkg_and_run
    return_value = function(pkg, kwargs)
  File "/opt/spack/lib/spack/spack/installer.py", line 1673, in build_process
    phase(pkg.spec, pkg.prefix)
  File "/opt/spack/lib/spack/spack/package.py", line 109, in phase_wrapper
    phase(spec, prefix)
  File "/opt/spack/var/spack/repos/builtin/packages/py-ipykernel/package.py", line 49, in install_data
  File "/opt/spack/lib/spack/spack/util/executable.py", line 147, in __call__
    quoted_args = [arg for arg in args if re.search(r'^"|^\'|"$|\'$', arg)]
  File "/opt/spack/lib/spack/spack/util/executable.py", line 147, in <listcomp>
    quoted_args = [arg for arg in args if re.search(r'^"|^\'|"$|\'$', arg)]
  File "/usr/lib/python3.6/re.py", line 182, in search
    return _compile(pattern, flags).search(string)
TypeError: expected string or bytes-like object
==> [2020-11-18-22:20:55.281313] Flagging py-ipykernel-5.3.4-wyn6znqdqfrlv6wklwntnensz4ycpoy3 as failed: TypeError: expected string or bytes-like object
==> [2020-11-18-22:20:55.316126] ChildError: TypeError: expected string or bytes-like object
```

--- a/var/spack/repos/builtin/packages/py-ipykernel/package.py
+++ b/var/spack/repos/builtin/packages/py-ipykernel/package.py
@@ -45,4 +45,5 @@ class PyIpykernel(PythonPackage):
 
     def install_data(self, spec, prefix):
         """ install the Jupyter kernel spec """
-        self.spec['python'].command('-m ipykernel', ['install'])
+        self.spec['python'].command(
+            '-m', 'ipykernel', 'install', '--prefix=' + prefix)

